livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py

```diff
@@ -161,6 +162,85 @@ def with_cerebras(
             temperature=temperature,
         )
 
+    @staticmethod
+    def with_vertexai(
+        *,
+        model: str | VertexModels = "google/gemini-1.5-pro",
+        project_id: str | None = None,
+        location: str | None = "us-central1",
+        client: Any | None = None,
+        user: str | None = None,
+        temperature: float | None = None,
+    ) -> LLM:
+        """
+        Create a new instance of VertexAI LLM.
+
+        `project_id` must be set to your VERTEXAI PROJECT ID, either using the argument or by setting
+        the `VERTEXAI_PROJECT_ID` environmental variable.
+        """
+
+        project_id = project_id or os.environ.get("VERTEXAI_PROJECT_ID")
+        if project_id is None:
+            raise ValueError(
+                "VERTEXAI_PROJECT_ID is required, either set project_id argument or set VERTEXAI_PROJECT_ID environmental variable"
+            )
+        location = location or os.environ.get("VERTEXAI_LOCATION")
+        if location is None:
+            raise ValueError(
+                "VERTEXAI_LOCATION is required, either set location argument or set VERTEXAI_LOCATION environmental variable"
+            )
+        _gac = os.environ.get("GOOGLE_APPLICATION_CREDENTIALS")
+        if _gac is None:
+            raise ValueError(
+                "`GOOGLE_APPLICATION_CREDENTIALS` environment variable is not set. please set it to the path of the service account key file."
+            )
+
+        try:
+            import google.auth
+            import google.auth.transport.requests
+        except ImportError:
+            raise ImportError(
+                "Google Auth dependencies not found. Please install with: `pip install livekit-plugins-openai[vertex]`"
+            )
+
+        class OpenAICredentialsRefresher:
+            def __init__(self, **kwargs: Any) -> None:
+                self.client = openai.AsyncClient(**kwargs, api_key="DUMMY")
+                self.creds, self.project = google.auth.default(
+                    scopes=["https://www.googleapis.com/auth/cloud-platform"]
+                )
+
+            def __getattr__(self, name: str) -> Any:
+                if not self.creds.valid:
+                    auth_req = google.auth.transport.requests.Request()
+                    self.creds.refresh(auth_req)
```

Comment:
Do you know if refresh is doing a network call?

Reply:
it probably should but I will check their implementation

Reply:
The issue if it is doing one is they it'll block the main thread since the request is sync

Reply:
google seems to support asyncio in their library: https://github.com/googleapis/google-auth-library-python/issues/1417